sympy/printing/fcode.py

```diff
@@ -129,6 +131,31 @@ def _lead(self):
         else:
             raise ValueError("Unknown source format: %s" % self._settings['source_format'])
 
+    def _print_Symbol(self, expr):
+        if self._settings['name_mangling'] == True:
+            if expr not in self.mangled_symbols:
+                name = expr.name
+                while name.lower() in self.used_name:
+                    name += '_'
+                self.used_name.append(name.lower())
+                if name == expr.name:
+                    self.mangled_symbols[expr] = expr
+                else:
+                    self.mangled_symbols[expr] = Symbol(name)
+
+            expr = expr.xreplace(self.mangled_symbols)
+
+        name = super(CodePrinter, self)._print_Symbol(expr)
+
+        if name in self.reserved_words:
```

Comment:
Is this part of this PR?

Reply:
yeah, as this method is overriding `_print_Symbol` of `CodePrinter` class, it should also perform checks done by that method.

Reply:
or rather there is no need of it because `name = super(CodePrinter, self)._print_Symbol(expr)` is calling it anyway ?

Reply:
@ylemkimonI just checked, overriding `print_Symbol` is indeed avoiding [this](https://github.com/sympy/sympy/blob/888f16155784747383083a8cd48a7f0089780899/sympy/printing/codeprinter.py#L325). So that check has to be added inside `FcodePrinter.print_Symbol` too.

Reply:
You have to pass the subclass as the first parameter of `super`, like `name = super(FCodePrinter, self)._print_Symbol(expr)`.